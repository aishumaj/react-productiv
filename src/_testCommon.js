const testTodo = {id:1, title:"hello", description: "test", priority:1};

const testTodos = [{id:1, title:"hello", description: "test", priority:2},
  {id:1, title:"top priority", description: "test2", priority:1},
  {id:1, title:"hello", description: "test3", priority:3}]

export {testTodo, testTodos};