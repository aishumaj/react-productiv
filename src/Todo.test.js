import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo"
import { testTodo } from "./_testCommon";

describe("productiv app", function () {
  it("renders without crashing", function () {
    render(<Todo todo={testTodo} />);
  });

  it("contains expected title and description", function () {
    const result = render(<Todo todo={testTodo}/>);
    expect(result.queryByText("hello")).toBeInTheDocument();
    expect(result.queryByText("test")).toBeInTheDocument();
  });

  test("matches snapshot", function () {
    const { container } = render(<Todo todo={testTodo}/>);
    expect(container).toMatchSnapshot();
  });

});