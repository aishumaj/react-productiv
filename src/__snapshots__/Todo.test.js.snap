// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`productiv app matches snapshot 1`] = `
<div>
  <div
    class="Todo"
    id="1"
  >
    <div>
      <b>
         
        hello
         
      </b>
       
      <small>
         (priority: 
        1
        )
      </small>
    </div>
    <div>
      <small>
        test
      </small>
    </div>
  </div>
</div>
`;
